librz/bin/p/bin_elf.inc

```diff
@@ -387,13 +387,323 @@ static ut64 reloc_targets_vfile_size(RzBinFile *bf, ELFOBJ *obj) {
 	return Elf_(rz_bin_elf_get_relocs_count)(obj) * reloc_target_size(obj);
 }
 
-static void patch_reloc(struct Elf_(rz_bin_elf_obj_t) * obj, RzBinElfReloc *rel, ut64 S, ut64 B, ut64 L) {
+static void patch_reloc(struct Elf_(rz_bin_elf_obj_t) * obj, RzBinElfReloc *rel, ut64 S, ut64 B, ut64 L, ut64 GOT) {
 	ut16 e_machine = obj->ehdr.e_machine;
-	ut64 val;
+	ut64 val = 0;
 	ut64 A = rel->addend, P = rel->vaddr;
 	ut64 patch_addr = rel->paddr != UT64_MAX ? rel->paddr : Elf_(rz_bin_elf_v2p_new)(obj, rel->vaddr);
 	ut8 buf[8];
 	switch (e_machine) {
+	case EM_QDSP6:;
+		ut8 word = 0;
```

Comment:
Can you please extract the reloc patching for Hexagon in a separate function? We are working on improving and refactoring relocations patching in ELF and it will make that easier in the future:
- https://github.com/rizinorg/rizin/pull/1699
- https://github.com/rizinorg/rizin/issues/1141